server.py

```diff
@@ -51,12 +51,13 @@ async def create_url(request: Request):
 
     try:
         alias = urljson.get('alias')
+        if not bool(re.match("^[a-zA-Z0-9]+$", alias)):
+            raise ValueError("alias must only contain alphanumeric characters")
         if alias is None:
             if args.disable_random_alias:
                 raise KeyError("alias must be specified")
             else:
                 alias = generate_alias(urljson['url'])
-
         with query_time.labels("create").time():
             if sqlite_helpers.insert_url(DATABASE_FILE, urljson['url'], alias):
                 url_count.inc(1)
```

Comment:
the checking should be done after the `None` block, like

```py
if alias is None:
  # existing code here...

if not alias.isalnum():
  raise ValueError("alias must only contain alphanumeric characters")
```

that way we will for sure have `alias` be a string. we also can skip regex usage, because python strings have a cool built in method for checking what chars exist, see the example below

```py
a = "asdf"
a.isalnum() # -> True

a = "asdf1"
a.isalnum() # -> True

a = "asdf./"
a.isalnum() # -> False
```

Reply:
Got it! I will fix it right now.

Reply:
Didn't even know that was a built in method.

Reply:
Same. Learning a new thing every day.